docs/csharp/language-reference/compiler-messages/cs8352.md

```diff
@@ -0,0 +1,122 @@
+---
+description: "Compiler Error CS8352"
+title: "Compiler Error CS8352"
+ms.date: 12/05/2023
+f1_keywords:
+  - "CS8352"
+helpviewer_keywords:
+  - "CS8352"
+---
+# Compiler Error CS8352
+
+Cannot use variable 'variable' in this context because it may expose referenced variables outside of their declaration scope.
+
+This error indicates potential undefined behavior when there's a risk of accessing a stack allocated variable after it has already been deleted from the stack.
+
+## To correct this error
+
+Ways to avoid `CS8352` are, for example:
+
+* Use the [`in`](../keywords/method-parameters.md#in-parameter-modifier) keyword and create the object in wider scope
+* Use the [`new`](../operators/new-operator.md) operator to create your object on a heap instead of a stack
+* Use [`ref`](../keywords/method-parameters.md#ref-parameter-modifier) keyword to notify the compiler of intentional object passing
+* Consider [`unsafe`](../keywords/unsafe.md) code to directly access memory.
```

Comment:
After reworking the examples, this doesn't match the examples.  How about this text instead:

```suggestion
You fix `CS8352`by ensuring the variable's scope is at least as wide as the variable that refers to it. You do this by either allocating the storage on heap, or in the stack on the calling context.
```